htdocs/compta/facture/card.php

```diff
@@ -471,9 +471,14 @@
 		}
 	} elseif ($action == 'setretainedwarrantydatelimit' && $usercancreate) {
 		$object->fetch($id);
-		$result = $object->setRetainedWarrantyDateLimit(GETPOSTFLOAT('retained_warranty_date_limit'));
-		if ($result < 0) {
-			dol_print_error($db, $object->error);
+		$retainedWarrantyDateLimit = GETPOST('retained_warranty_date_limit');
+		if (preg_match('/^\d{4}-\d{2}-\d{2}$/', $retainedWarrantyDateLimit)) {
```

Comment:
We should never used such things when reading date, format vary on language..
Can you try
$retainedWarrantyDateLimit = GETPOSTDATE('retained_warranty_date_limit');

and test if value is empty or not